projects/laji/src/app/shared/navbar/navbar.component.ts

```diff
@@ -54,7 +55,9 @@ export class NavbarComponent implements OnInit, OnDestroy {
     private ngZone: NgZone
   ) {
     this.navId = environment.type + '-nav';
-    this.devRibbon = !environment.production || environment.type === Global.type.beta;
+    this.devRibbon = environment.type === Global.type.vir ?
+      virEnvironment.displayDevRibbon :
+      !environment.production || environment.type === Global.type.beta;
```

Comment:
I think the `displayDevRibbon` should have highest priority since it's explicitly stating that the ribbon should be shown. Now the flag is checked for `vir` environment. Also, you're not supposed to check the `vir` project's environment file directly. The environment should be always imported from `projects/laji/src/environments/environment.ts` in code. The build processes handle rewriting that file (see `angular.json`, it contains the file replacement logic).

Proposal:

1. Set `devRibbon` for beta env, so the `environment.type === Global.type.beta` check can be removed.
2. Change the logic to:
```
this.devRibbon = environment.displayDevRibbon ||  environment.production === false
```

As a result, the `displayDevRibbon` is shown when explicitly stated in the environment. Checking a falsy `environment.production` is also OK, because it should never be `false` in production environment.

Reply:
Thanks for the feedback! I have one question: How to avoid warnings when referring to "environment.displayDevRibbon" in step 2 of the proposal? TypeScript does not seem to be aware that the content of environment.ts gets rewritten, it just sees that displayDevRibbon cannot be found in that file directly.

Reply:
That can be fixed by adding typings for the base environment (`projects/laji/src/environments/environment.ts`), or adding `displayDevRibbon` to the base environment properties so TS can infer the type correctly. The latter option would be less work? I think `displayDevRibbon: false` would make sense since it's supposed to be false by default.